src/System.Text.Primitives/System/Text/Parsing/InvariantUtf16/InvariantUtf16_Int16.cs

```diff
@@ -0,0 +1,75 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Diagnostics;
+using System.Text.Utf8;
+
+namespace System.Text
+{
+    public static partial class PrimitiveParser
+    {
+        public static partial class InvariantUtf16
+        {
+            public unsafe static bool TryParseInt16(char* text, int length, out short value)
+            {
+                int consumed;
+                var span = new ReadOnlySpan<byte>(text, length * 2);
```

Comment:
@botaberg, please change the PR to use sizeof(char). It's a great suggestion. Once you update the PR I will merge. We can tweak the hex parsing separately.